reconcile/slack_cluster_usergroups.py

```diff
@@ -75,42 +85,56 @@ def get_desired_state(slack):
                       if include_user(u, cluster_name, cluster_users)]
         users = slack.get_users_by_names(user_names)
         channels = slack.get_channels_by_names([slack.chat_kwargs['channel']])
-        desired_state.append({
+        desired_state.setdefault(slack.workspace_name, {})[usergroup] = {
             "workspace": slack.workspace_name,
             "usergroup": usergroup,
             "usergroup_id": ugid,
             "users": users,
             "channels": channels,
             "description": f'Users with access to the {cluster_name} cluster',
-        })
+        }
 
     return desired_state
 
 
 def get_current_state(slack, usergroups):
-    current_state = []
+    """
+    Get the current state of the Slack cluster usergroups.
+
+    :param slack: client for calling Slack API
+    :type slack: reconcile.utils.slack_api.SlackApi
+
+    :param usergroups: cluster usergroups to get state of
+    :type usergroups: Iterable
+
+    :return: current state data, keys are workspace -> usergroup
+                (ex. state['coreos']['app-sre-ic']
+    :rtype: dict
+    """
+    current_state = {}
 
     for ug in usergroups:
         users, channels, description = slack.describe_usergroup(ug)
-        current_state.append({
+        current_state.setdefault(slack.workspace_name, {})[ug] = {
             "workspace": slack.workspace_name,
             "usergroup": ug,
             "users": users,
             "channels": channels,
             "description": description,
-        })
+        }
 
     return current_state
 
 
 def run(dry_run):
     slack = init_slack_workspace(QONTRACT_INTEGRATION)
     desired_state = get_desired_state(slack)
-    usergroups = [d['usergroup'] for d in desired_state]
+    usergroups = []
+    for _, workspace_state in desired_state.items():
```

Comment:
This is easier as
```suggestion
    for workspace_state in desired_state.values()
```

Reply:
That's a fair point.  I'll consider that pattern in the future when I'm throwing away the other value.  Thanks!